pkg/client/unversioned/conditions.go

```diff
@@ -28,73 +28,77 @@ import (
 	"k8s.io/kubernetes/pkg/watch"
 )
 
-// ControllerHasDesiredReplicas returns a condition that will be true if and only if
-// the desired replica count for a controller's ReplicaSelector equals the Replicas count.
-func ControllerHasDesiredReplicas(c Interface, controller *api.ReplicationController) wait.ConditionFunc {
+// ReplicationControllerHasDesiredReplicas returns true if and only if the
+// desired replica count for a controller's ReplicaSelector equals the Replicas count.
+func ReplicationControllerHasDesiredReplicas(event watch.Event) (bool, error) {
+	switch event.Type {
+	case watch.Deleted:
+		return false, errors.NewNotFound(unversioned.GroupResource{Resource: "replicationcontrollers"}, "")
+	}
 
-	// If we're given a controller where the status lags the spec, it either means that the controller is stale,
-	// or that the rc manager hasn't noticed the update yet. Polling status.Replicas is not safe in the latter case.
-	desiredGeneration := controller.Generation
+	switch t := event.Object.(type) {
+	case *api.ReplicationController:
+		// If we're given a controller where the status lags the spec, it either means that the controller is stale,
+		// or that the rc manager hasn't noticed the update yet. Polling status.Replicas is not safe in the latter case.
+		desiredGeneration := t.Generation
```

Comment:
This is used from a call site that needs a specific generation passed in.  I'm fine with just passing the generation (not the entire controller), but the callsite needs to specify what generation he cares about.  Otherwise he can get stuck waiting longer than he needs to.